types/src/extended_header.rs

```diff
@@ -198,4 +272,45 @@ mod tests {
             ))
         ));
     }
+
+    #[test]
+    fn verify() {
+        let eh_block_1 = sample_eh_block_1();
+        let eh_block_27 = sample_eh_block_27();
+
+        eh_block_1.verify(&eh_block_27).unwrap();
+    }
+
+    #[test]
+    fn verify_invalid_height() {
+        let eh_block_27 = sample_eh_block_27();
+        eh_block_27.verify(&eh_block_27).unwrap_err();
+    }
+
+    #[test]
+    fn verify_invalid_chain_id() {
+        let eh_block_1 = sample_eh_block_1();
+        let mut eh_block_27 = sample_eh_block_27();
+
+        eh_block_27.header.chain_id = "1112222".parse::<Id>().unwrap();
```

Comment:
```suggestion
        eh_block_27.header.chain_id = "1112222".parse().unwrap();
```
isn't type redundant?